src/itemdef.cpp

```diff
@@ -543,7 +543,7 @@ class CItemDefManager: public IWritableItemDefManager
 					request = m_get_clientcached_queue.pop();
 
 			m_get_clientcached_queue.pushResult(request,
-					createClientCachedDirect(request.key, gamedef));
+					createClientCachedDirect(request.key, (Client *)gamedef));
```

Comment:
in fact no, at this moment it's not possible to change this because we depend on IWritableItemDefManager which is client & server side